test_requests.py

```diff
@@ -188,6 +188,30 @@ def test_HTTP_302_ALLOW_REDIRECT_GET(self, httpbin):
         assert r.history[0].status_code == 302
         assert r.history[0].is_redirect
 
+    def test_HTTP_302_TOO_MANY_REDIRECTS(self, httpbin):
+        try:
+            requests.get(httpbin('relative-redirect', '50'))
+        except TooManyRedirects as e:
+            url = httpbin('relative-redirect', '20')
```

Comment:
Checked some other tests, and found they were just using `httpbin` to generate URLs for assertion.  I think it's good that if the `httpbin` API changes, that this test will break, as opposed to silently failing.  Also it provides further testing that the redirection logic is working.